### ISSUE TITLE
IteratorOfPartitionsSet runs out of memory

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#4515]
This worked the day before yesterday and does not work since yesterday - perhaps after #4505? 

Happens when all packages are loaded:
```
########> Diff in /circa/scratch/gap-jenkins/workspace/GAP-pkg-update-master-q\
uicktest/GAPCOPTS/64build/GAPGMP/gmp/GAPTARGET/standard/label/kovacs/GAP-pkg-u\
pdate-master-snapshot/tst/testinstall/combinat.tst:474
# Input is:
for s in [[], [5], [1,2,3,4], [2,5,7], ["a","b","c","d","e"], [3..13]] do
     pn1:= PartitionsSet( s );
     pn2:= List( IteratorOfPartitionsSet( s ) );
     if Length(pn1) <> Length(pn2) then
       Error( "wrong number of elements" );
     elif Set(pn1) <> Set(pn2) then
       Error( "different elements" );
     fi;
   od;
# Expected output:
# But found:
Error, reached the pre-set memory limit
(change it with the -o command line option)
########
```
but only in the GAP master branch - works fine in stable-4.11.

Could be of course a simple overflow due to many packages being loaded.

**Comment by fingolfin** [comment]
In PR #4502 I changed the `[3..13]` to `[3..9]` which I think should address this issue, but I can't test it, so... It also makes the test *faster*

**Comment by fingolfin** [comment]
Should be resolved by PR #4502
